src/runtime/auto-prompt-manager.ts

```diff
@@ -813,28 +973,56 @@ export class AutoPromptManager {
     return Math.floor(nano / Math.pow(10, 6));
   }
 
-  /**
-   * Checks AudienceAction eligbility, used to filter potential actions.
-   */
-  private checkActionEligibility_(
-    actionType: string,
-    timestamps: ActionsTimestamps
-  ): boolean {
-    if (actionType === TYPE_REWARDED_SURVEY) {
-      const isAnalyticsEligible =
-        GoogleAnalyticsEventListener.isGaEligible(this.deps_) ||
-        GoogleAnalyticsEventListener.isGtagEligible(this.deps_) ||
-        GoogleAnalyticsEventListener.isGtmEligible(this.deps_);
-      if (!isAnalyticsEligible) {
-        return false;
-      }
-      // Do not show survey if there is a previous completion record.
-      // Client side eligibility is required to handle identity transitions
-      // after sign-in flow. TODO(b/332759781): update survey completion check
-      // to persist even after 2 weeks.
-      return !(timestamps[TYPE_REWARDED_SURVEY]?.completions || []).length;
+  private getPromptFrequencyCapDuration_(
+    frequencyCapConfig: FrequencyCapConfig,
+    interventionOrchestration: InterventionOrchestration
+  ): Duration | undefined {
+    const swgDuration = interventionOrchestration.promptFrequencyCap?.duration;
+
+    if (!swgDuration) {
+      this.eventManager_.logSwgEvent(
+        AnalyticsEvent.EVENT_PROMPT_FREQUENCY_CONFIG_NOT_FOUND
+      );
+      return frequencyCapConfig.anyPromptFrequencyCap?.frequencyCapDuration;
+    }
+    return this.convertSwgDurationToSeconds_(swgDuration);
+  }
+
+  private getGlobalFrequencyCapDuration_(
+    frequencyCapConfig: FrequencyCapConfig,
+    interventionFunnel: InterventionFunnel
+  ): Duration | undefined {
+    const swgDuration = interventionFunnel.globalFrequencyCap?.duration;
+    return swgDuration
+      ? this.convertSwgDurationToSeconds_(swgDuration)
```

Comment:
Same as above, if the provided globalFrequencyCap is present but invalid, then the current logic is to NOT frequency cap the prompt. Should we revisit this logic to fallback to the default?